core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1646,7 +1646,7 @@ class SparkContext(config: SparkConf) extends Logging {
   def killExecutors(executorIds: Seq[String]): Boolean = {
     schedulerBackend match {
       case b: CoarseGrainedSchedulerBackend =>
-        b.killExecutors(executorIds, replace = false, force = true).nonEmpty
+        b.killExecutors(executorIds, replace = false, force = false).nonEmpty
```

Comment:
Good point. I thought about to add a new API, but it needs quite some changes. The issue is that in ExecutorAllocationManager#removeExecutors, we need a way to tell client.killExecutors(executorIdsToBeRemoved) to skip those busy executors. We can either add one new API to SparkContext, or call CoarseGrainedSchedulerBackend instead SparkContext in ExecutorAllocationManager. What do you suggest?